### ISSUE TITLE
Caffe compilation error: cuda_compile_generated_math_functions.cu.o.cmake:264

### ISSUE THREAD

**Issue by danielmorozoff** [BVLC/caffe#2862]
Using CUDA 7 on osx 10.10.4

Get compilaiton error at: 
[  2%] Building NVCC (Device) object src/caffe/CMakeFiles/cuda_compile.dir/util/cuda_compile_generated_math_functions.cu.o

Also- weirdly I have cudnn placed in my /usr/local/cuda/  dir but it does not get detected. (cudnn.h in include & lib files in lib). 
Set env variables in zsh shell : 
export PATH=$PATH:/usr/local/cuda/bin
export LD_LIBRARY_PATH=/usr/local/cuda/lib

Installed all dependencies- see cmake output bellow: 

The C compiler identification is AppleClang 6.1.0.6020053
-- The CXX compiler identification is AppleClang 6.1.0.6020053
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   thread
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Found GFlags: /usr/local/include
-- Found gflags  (include: /usr/local/include, library: /usr/local/lib/libgflags.a)
-- Found Glog: /usr/local/include
-- Found glog    (include: /usr/local/include, library: /usr/local/lib/libglog.dylib)
-- Found PROTOBUF: /usr/local/lib/libprotobuf.dylib
-- Found PROTOBUF Compiler: /usr/local/bin/protoc
-- Found HDF5: /opt/local/lib/libhdf5_hl.dylib;/opt/local/lib/libhdf5.dylib
-- Found LMDB: /opt/local/include
-- Found lmdb    (include: /opt/local/include, library: /opt/local/lib/liblmdb.dylib)
-- Found LevelDB: /opt/local/include
-- Found LevelDB (include: /opt/local/include, library: /opt/local/lib/libleveldb.dylib)
-- Found Snappy: /opt/local/include
-- Found Snappy  (include: /opt/local/include, library: /opt/local/lib/libsnappy.dylib)
-- CUDA detected: 7.0
-- Automatic GPU detection failed. Building for all known architectures.
-- Added CUDA NVCC flags for: sm_20 sm_21 sm_30 sm_35 sm_50
-- OpenCV found (/opt/local/share/OpenCV)
-- Found vecLib: /System/Library/Frameworks/Accelerate.framework/Versions/Current/Frameworks/vecLib.framework/Versions/Current/Headers
-- Found vecLib as part of Accelerate.framework
-- Found PythonInterp: /usr/local/bin/python2.7 (found suitable version "2.7.10", minimum required is "2.7")
-- Found PythonLibs: /usr/lib/libpython2.7.dylib (found suitable version "2.7.10", minimum required is "2.7")
-- Found NumPy: /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/core/include (found suitable version "1.9.2", minimum required is "1.7.1")
-- NumPy ver. 1.9.2 found (include: /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/core/include)
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   python
-- Found Doxygen: /opt/local/bin/doxygen (found version "1.8.9.1")
-- Detected Doxygen OUTPUT_DIRECTORY: ./doxygen/
-- Found Git: /usr/bin/git (found version "2.3.2 (Apple Git-55)")
--
-- ******************* Caffe Configuration Summary *******************
-- General:
--   Version           :   <TODO> (Caffe doesn't declare its version in headers)
--   Git               :   rc2-368-gd4aa5fe
--   System            :   Darwin
--   C++ compiler      :   /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
--   Release CXX flags :   -O3 -DNDEBUG -fPIC -Wall -Wno-sign-compare -Wno-uninitialized
--   Debug CXX flags   :   -g -fPIC -Wall -Wno-sign-compare -Wno-uninitialized
--   Build type        :   Release
--
--   BUILD_SHARED_LIBS :   ON
--   BUILD_python      :   ON
--   BUILD_matlab      :   OFF
--   BUILD_docs        :   ON
--   CPU_ONLY          :   OFF
--
-- Dependencies:
--   BLAS              :   Yes (vecLib)
--   Boost             :   Yes (ver. 1.58)
--   glog              :   Yes
--   gflags            :   Yes
--   protobuf          :   Yes (ver. 3.0.0)
--   lmdb              :   Yes (ver. 0.9.14)
--   Snappy            :   Yes (ver. 1.1.1)
--   LevelDB           :   Yes (ver. 1.9)
--   OpenCV            :   Yes (ver. 3.0.0)
--   CUDA              :   Yes (ver. 7.0)
--
-- NVIDIA CUDA:
--   Target GPU(s)     :   Auto
--   GPU arch(s)       :   sm_20 sm_21 sm_30 sm_35 sm_50
--   cuDNN             :   Not found
--
-- Python:
--   Interpreter       :   /usr/local/bin/python2.7 (ver. 2.7.10)
--   Libraries         :   /usr/lib/libpython2.7.dylib (ver 2.7.10)
--   NumPy             :   /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/core/include (ver 1.9.2)
--
-- Documentaion:
--   Doxygen           :   /opt/local/bin/doxygen (1.8.9.1)
--   config_file       :   /Users/danielmorozoff/Desktop/Git-Repos/caffe/.Doxyfile
--
-- Install:
--   Install path      :   /Users/danielmorozoff/Desktop/Git-Repos/caffe/build/install
--
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/danielmorozoff/Desktop/Git-Repos/caffe/build

**Comment by danielmorozoff** [comment]
Seems like I got the compiler to run a bit longer by using the standard make - editing the INCLUDE_DIRS & LIBRARY_DIRS

then it pooped out at NVCC :
NVCC src/caffe/layers/absval_layer.cu
/opt/local/include/boost/smart_ptr/detail/sp_counted_base_clang.hpp(27): error: explicit type is missing ("int" assumed)

is this a problem with boost 1.58.0?

**Comment by danielmorozoff** [comment]
boost 1.58 is the problem. seems like boost 1.57 should be ok.
if you used homebrew check this out:
http://itinerantbioinformaticist.blogspot.com/2015/05/caffe-incompatible-with-boost-1580.html

since I am using macports 2.3.3, the last port with boost 1.57 was 135230. So do the following commands:

cd /tmp
svn co http://svn.macports.org/repository/macports/trunk/dports/devel/boost --revision 135230
cd boost
sudo port install

*check installed boost versions*
port installed boost
sudo port activate boost @1.57.0_1+no_single+no_static+python27

Now compilation proceeds-- new problem after a while:

LD -o .build_release/lib/libcaffe.so
clang: warning: argument unused during compilation: '-pthread'
Undefined symbols for architecture x86_64:

this seems to have been addressed here: https://github.com/BVLC/caffe/issues/2018
Will check back on this to see if I can get it running

**Comment by danielmorozoff** [comment]
need to run make all with proper flags.
 LDFLAGS="`pkg-config --libs protobuf` `pkg-config --libs opencv`" make all

Seems to compile...

**Comment by danielmorozoff** [comment]
I guess you guys can close this issue.

**Comment by Benedetta89** [comment]
I did what you said and I got this error  
CMake Error at CMakeLists.txt:28 (find_package):
  By not providing "FindMATIO.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "MATIO", but
  CMake did not find one.

  Could not find a package configuration file provided by "MATIO" with any of
  the following names:

    MATIOConfig.cmake
    matio-config.cmake

  Add the installation prefix of "MATIO" to CMAKE_PREFIX_PATH or set
  "MATIO_DIR" to a directory containing one of the above files.  If "MATIO"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/benedetta/Scrivania/caffe/build/CMakeFiles/CMakeOutput.log".
See also "/home/benedetta/Scrivania/caffe/build/CMakeFiles/CMakeError.log".
Makefile:544: set di istruzioni per l'obiettivo "cmake_check_build_system" non riuscito
make: *** [cmake_check_build_system] Errore 1

why?
